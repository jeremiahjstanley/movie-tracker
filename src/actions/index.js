export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
});

export const signUp = (username, password, email) => ({
  type: 'SIGN_UP',
  username,
  password, 
  email
});

export const logIn = (username, password) => ({
  type: 'LOG_IN',
  username,
  password
});

export const logOut = (username) => ({
  type: 'LOG_OUT',
  username
});

