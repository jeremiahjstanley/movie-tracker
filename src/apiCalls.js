
const fetchMovieData = async (key) => {
  const url = `https://api.themoviedb.org/3/movie/550?api_key=${key}`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
};

export default fetchMovieData;