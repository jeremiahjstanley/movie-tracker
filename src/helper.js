export const movieCleaner = results => {
  return {
    title: results.original_title,
    releaseDate: results.release_date,
    rating: results.vote_average,
    tagline: results.tagline,
    movieImage: results.poster_path,
    overview: results.overview,
    favorite: false
  };
};