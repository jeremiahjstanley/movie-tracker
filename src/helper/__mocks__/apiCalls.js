export const fetchMovieData = jest.fn();

export const deleteFavoriteFromDatabase = jest.fn();

export const getFavoritesFromDatabase = jest.fn().mockImplementation(() => ({
  data: [{title: 'Con Air'}, {title: 'Face/Off'}, {title: 'Captain Ron'}]
}))

export const createUser = jest.fn().mockImplementation(() => ({
  id: 2
}));

export const sendFavoriteToDatabase = jest.fn()
