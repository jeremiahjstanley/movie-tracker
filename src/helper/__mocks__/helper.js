export const movieCleaner = jest.fn()