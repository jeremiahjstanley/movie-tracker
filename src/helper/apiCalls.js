const fetchMovieData = async (key) => {
  const url = `https://api.themoviedb.org/3/discover/movie?api_key=${key}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
};

export default fetchMovieData;