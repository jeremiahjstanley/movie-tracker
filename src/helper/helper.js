export const movieCleaner = response => {
  const movies = response.results.map(result => ({
    id: result.id,
    backdrop_path: result.backdrop_path,
    poster_path: result.poster_path,
    overview: result.overview,
    vote_average: result.vote_average,
    release_date: result.release_date,
    tagline: result.tagline,
    title: result.original_title
  })
  );
  return movies;
};