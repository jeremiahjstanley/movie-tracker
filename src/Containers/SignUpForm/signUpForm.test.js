import React from 'react';
import { SignUpForm, mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow } from 'enzyme';
import { createUser } from '../../helper/apiCalls';
import { signUp } from '../../actions';

jest.mock('../../helper/apiCalls');


describe('SignUp Form tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignUpForm />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state of email, name, password, and errorMessage all empty strings', () => {
    const expected = {
      email: '',
      name: '',
      password: '',
      errorMessage: ''
    };

    expect(wrapper.state()).toEqual(expected);
  });

  it('should update the state when handleChange is invoked', () => {
    const mockEvent = { target: { value: 'Nick Cage', name: 'name' } };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toBe('Nick Cage');
  });

  it('should invoke handleSubmit when the form is submitted', () => {
    const spy = spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.setState({
      email: 'nick@msn.com',
      name:'nick',
      password: 'unleashthecage'
    });

    const mockEvent = { preventDefault: jest.fn() };

    const submitButton = wrapper.find('form');
    submitButton.simulate('submit', mockEvent);

    expect(spy).toHaveBeenCalledWith(mockEvent);
  });

  it('should invoke createUser when form is submitted', async () => {
    const mockSubmitForm = jest.fn();
    wrapper = shallow(<SignUpForm submitForm={mockSubmitForm}/>);
    wrapper.setState({
      email: 'nick@msn.com',
      name: 'nick',
      password: 'unleashthecage'
    });
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().handleSubmit(mockEvent);

    expect(createUser).toHaveBeenCalledWith('nick', 'nick@msn.com', 'unleashthecage');
  });

  it('should invoke submitForm when the form is submitted', async () => {
    const mockSubmitForm = jest.fn();
    wrapper = shallow(<SignUpForm submitForm={mockSubmitForm} />);
    wrapper.setState({
      email: 'nick@msn.com',
      name: 'nick',
      password: 'unleashthecage'
    });
    const mockEvent = { preventDefault: jest.fn() };

    await wrapper.instance().handleSubmit(mockEvent);
    expect(mockSubmitForm).toHaveBeenCalled();
  });

  it('should reset state to empty strings after form is submitted', async () => {
    const mockSubmitForm = jest.fn();
    const expected = {name: '', email: '', password: '', errorMessage: ''};
    wrapper = shallow(<SignUpForm submitForm={mockSubmitForm} />);
    wrapper.setState({
      email: 'nick@msn.com',
      name: 'nick',
      password: 'unleashthecage'
    });
    const mockEvent = { preventDefault: jest.fn() };

    await wrapper.instance().handleSubmit(mockEvent);
    const results = wrapper.state();

    expect(results).toEqual(expected);
  });

  describe('mapStateToProps', () => {
    it('should return an object with the email, name and favorites array', () => {
      const mockState = { login: { email: 'nickcage@aol.com', name: 'Nick Cage' } };
      const expected = { email: 'nickcage@aol.com', name: 'Nick Cage' };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch when the form is submitted to log the user in', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = signUp('nickcage@aol.com', 'Nick', 1);
      mappedProps.submitForm('nickcage@aol.com', 'Nick', 1);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});