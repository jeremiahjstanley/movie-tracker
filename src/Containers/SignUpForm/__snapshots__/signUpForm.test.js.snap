// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Form tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="exit-form"
          replace={false}
          to="/"
>
          X
</Link>,
        <h1
          className="form-header"
>
          Sign Up
</h1>,
        <input
          aria-label="Please Enter Your Name"
          className="name-field"
          name="name"
          onChange={[Function]}
          placeholder="name"
          type="text"
          value=""
/>,
        <input
          aria-label="Please Enter Your Email"
          className="email-input"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
/>,
        <input
          aria-label="Please Enter Your Password"
          className="password-input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          aria-label="Sign up for account"
>
          Sign Up!
</button>,
        <h3>
           
          
           
</h3>,
        <Link
          replace={false}
          to="/login"
>
          Have an exisiting account? Login here.
</Link>,
      ],
      "className": "form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "X",
          "className": "exit-form",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "X",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "form-header",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Please Enter Your Name",
          "className": "name-field",
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Please Enter Your Email",
          "className": "email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Please Enter Your Password",
          "className": "password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Sign up for account",
          "children": "Sign Up!",
        },
        "ref": null,
        "rendered": "Sign Up!",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "",
          " ",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Have an exisiting account? Login here.",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Have an exisiting account? Login here.",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="exit-form"
            replace={false}
            to="/"
>
            X
</Link>,
          <h1
            className="form-header"
>
            Sign Up
</h1>,
          <input
            aria-label="Please Enter Your Name"
            className="name-field"
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
/>,
          <input
            aria-label="Please Enter Your Email"
            className="email-input"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
/>,
          <input
            aria-label="Please Enter Your Password"
            className="password-input"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            aria-label="Sign up for account"
>
            Sign Up!
</button>,
          <h3>
             
            
             
</h3>,
          <Link
            replace={false}
            to="/login"
>
            Have an exisiting account? Login here.
</Link>,
        ],
        "className": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "X",
            "className": "exit-form",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "X",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "form-header",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Please Enter Your Name",
            "className": "name-field",
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Please Enter Your Email",
            "className": "email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Please Enter Your Password",
            "className": "password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Sign up for account",
            "children": "Sign Up!",
          },
          "ref": null,
          "rendered": "Sign Up!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "",
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Have an exisiting account? Login here.",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Have an exisiting account? Login here.",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
